Switch to a more accessible color scale

wchargin opened the pull request:
Summary:
Per @jart's request, I've been looking into color schemes that are more
accessible to people with deuteranomaly/deuteranopia (like me). Our
previous color scheme had three particularly bad color pairs:
yellow/lime, green/red (duh), and, to a lesser extent, blue/purple. [1]
I adjusted the first two of these pairs such that I could more
comfortably tell them apart at the scales at which they're used, but
the resulting scheme is still not great: the colors are just too
similar.

So I switched to the palette preferred by Paul Tol for
categorical data. (We already have several palettes of his and
modifications thereof in our `palettes.ts`, but not the best one!)

This image compares the three stages:
![Results of my comparison](https://user-images.githubusercontent.com/4317806/28694991-6d9e7d98-72e1-11e7-9fe3-581b7e32a1d2.png)

[1]: This pair is still pretty bad, just not as bad as the other two.

Here's what the resulting TensorBoard looks like:
![The histogram dashboard with the new color palette](https://user-images.githubusercontent.com/4317806/28695189-9f6df168-72e2-11e7-92e7-6fef5578d102.png)
![The scalar dashboard with the new color palette](https://user-images.githubusercontent.com/4317806/28695191-a053ad2a-72e2-11e7-987e-ab2554baa8a8.png)

The colors aren't always as vibrant as before on the scalar dashboard,
but I still think that they're attractive, especially when used at
larger scales (e.g., the histogram dashboard is quite pretty).
More importantly---they're more functional.

Test Plan:
Ask a colorblind person what they think!

In particular, the following is an interesting heuristic: look at the
list of checkboxes in the runs selector. Move from top to bottom, and
stop when you think you've gotten to a color that you've already seen.
If you have a false positive, the color palette is bad; this is very
much the case with all three bad pairs in the original palette, and also
(for me) with the blue/purple pair in the Mark II palette, but Tol's
palette has clearly distinct colors with no such false positives.

wchargin-branch: improve-color-a11y

wchargin commented:
Sure thing.

wchargin commented:
(Thanks!)

shancarter commented:
LGTM

wchargin commented:
Great; thanks @shancarter.

@dandelionmane, what would you like to see before merging this, if you'd like to merge it at all?

dandelionmane commented:
I'd be excited to put in the orange as the first color, and maybe take out lime. But it has my approval either way.

wchargin commented:
Updated to swap out the lime color for TensorBoard orange. I've emailed Paul to check whether this will impact the a11y. Updated screenshots here: http://imgur.com/a/ZGabM

wchargin commented:
Paul was kind enough to do a detailed analysis. Here is his response, posted with permission:

> Hi William,
> 
> I've examined your suggestion. There are two problems with orange
> #FF7043, one minor and one major:
> 1) It is a bit out of CMYK gamut. If I convert from RGB to CMYK and back
> again, I get #EE6F43. This means that a printout will probably look a
> bit duller than the screen. However, since the orange is the signature
> colour of TensorBoard, a slight shift in colour on a printout is not
> important enough.
> 2) According to my calculation, the distance to the green is 18 in
> deuteranope vision, but only 4 in protanope vision, which is far too
> close. All distances in my scheme are at least 14. It is a nice example
> where there is a significant difference between protanope and
> deuteranope vision.
> 
> To have some sort of green, it has to be shifted to blue. I could manage
> a distance of 10 with #227744. However, the other colours do not combine
> well esthetically with the orange. Especially the orange and red clash.
> So I redesigned the scheme starting with the orange and getting colours
> of roughly the same "vividness", slightly more vibrant as you called it.
> The smallest distance is now 15, except 13 between grey and teal in
> protanope vision, which is not significantly smaller. I've attached the
> scheme. By the way, I have corrected the decimal code of orange in your
> diagram. Also attached is a test map, in which the left section contains
> all colours, while each of the other sections contains one colour with
> one cell of each other colour.
> 
> I give the colours in order of hue, but if you prefer a 'random' order
> that still looks ordered: orange, blue, red, cyan, magenta, teal, grey.
> The odd colours are then hot, the even colours cool. Teal may just look
> like dark grey to you, but it is definitely a colour.
> 
> The new scheme should be distinct enough. I hope it also works
> esthetically in TensorBoard plots, for all people.
> 
> Best regards, Paul

Attachments:

[tensorboard_adjusted.pdf](https://github.com/tensorflow/tensorboard/files/1209089/tensorboard_adjusted.pdf)
[tensorboard_adjusted_map.pdf](https://github.com/tensorflow/tensorboard/files/1209093/tensorboard_adjusted_map.pdf)

Raster versions:
![tensorboard_adjusted](https://user-images.githubusercontent.com/4317806/29086500-ad86ffb2-7c27-11e7-8d46-8a81d9a56f3a.png)
![tensorboard_adjusted_map](https://user-images.githubusercontent.com/4317806/29086503-b073d01a-7c27-11e7-8477-c31659280d01.png)

Will update the PR now.
